src/Css/Style.php

```diff
@@ -1446,6 +1493,16 @@ public function set_prop(string $prop, $val, bool $important = false, bool $clea
                 $val = self::$_defaults[$prop];
             }
 
+            // Do not resolve CSS custom property values when setting them,
+            // because the required custom property might not have been set yet,
+            // or need to be derived from a parent. Let the compute() method
+            // resolve this when the value is needed. Just set the property now.
+            if (is_string($val) && strpos($val, 'var(') !== false) {
+                $this->_props[$prop] = $val;
```

Comment:
Setting the specified value here is problematic due to assumptions made elsewhere in the code. Normally when a property has an invalid computation we drop the specified value entirely. Which is fine since the value must be valid when set, and the default value is used otherwise.

CSS variables may not be valid at the time they are set. Or may contain invalid values once they are resolved. But we want to maintain the specified value in case the variable resolves correctly later on.

The following example will trigger an error because when Dompdf retrieves the font size it returns an unexpected value (null):

```
<style>
    div {
        font-size: var(--bad);
    }
</style>
<div>dompdf</div>
```

I think the resolution might be to modify the `computed` method to recompute the property if computing with the specified value returns null.